File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    
    priorities = priorities.map((priority, index) => [index, priority]);
    
    let cnt = 0;
    while (priorities.length) {
        if (priorities.some(item => item[1] > priorities[0][1])) {
            // 더 높은 우선순위 있으면 뒤로 보냄
            priorities.push(priorities.shift());
        } else { // 가장 높은 우선순위면 바로 실행
            let tmp = priorities.shift();
            cnt++;
            if (tmp[0] === location) return cnt;
        }
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
import heapq
from collections import deque

def solution(priorities, location):
    wq = deque()
    pq = []
    
    for idx, priority in enumerate(priorities):
        wq.append((idx, priority))
        heapq.heappush(pq, -priority) # negative value for descending order of absolute value
        
    answer = 0
    while True:
        if wq[0][1] < -pq[0]: # revert to positive value and compare
            wq.append(wq.popleft())
        else:
            num = wq.popleft()[0]
            heapq.heappop(pq)
            answer += 1
            if num == location:
                break
    
    return answer